### PR TITLE
Improve package load time

### PR DESCRIPTION
**Pull request by olmokramer**
Moves `require 'github'` and `require './package-manager'` to package activation, making the package load faster.

### CODE REVIEW

**Comment by jerone** [comment]
:+1:

**Comment by groupsky** [comment]
Related to #33

**Comment by groupsky** [comment]
@olmokramer can you updated with latest master and resubmit. The travis builds are starting to work for PRs but this one seems to be broken.

File: lib/sync-settings.coffee [hunk 1]
```diff
@@ -1,9 +1,8 @@
 # imports
 {BufferedProcess} = require 'atom'
-GitHubApi = require 'github'
-_ = require 'underscore-plus'
-PackageManager = require './package-manager'
 fs = require 'fs'
+_ = require 'underscore-plus'
+[GitHubApi, PackageManager] = []
```
Comment (groupsky) [inline lib/sync-settings.coffee]: Will you add some explanation as comment to make it clear why the import is actually during package activation

Comment (olmokramer) [inline lib/sync-settings.coffee]: It does that so Atom doesn't have to read the module files when it starts up. It reduced package load time by 10-20ms on my machine (according to timecop).

**Comment by olmokramer** [comment]
Ok I merged the master branch, but Travis is still failing...

**Comment by groupsky** [comment]
It seems the changes [broke the specs](https://travis-ci.org/Hackafe/atom-sync-settings/builds/66184328#L133). Maybe the specs should pre-call activate?

File: lib/sync-settings.coffee (hunk 1 above)
Comment (groupsky) [inline lib/sync-settings.coffee]: I meant to add it in the code, as after the pull is merged it won't be so obvious why the imports are moved

Comment (olmokramer) [inline lib/sync-settings.coffee]: Ah, right. Duh... :p
I have some time this weekend to work on this.

**Comment by olmokramer** [comment]
Yeah I think so. Should be fairly easy to fix :)

**Comment by groupsky** [comment]
@olmokramer I made a PR to your repo fixing the specs

**Comment by olmokramer** [comment]
Thanks for that! I merged the changes, and the specs complete successfully (at least on my machine).

File: lib/sync-settings.coffee [hunk 2]
```diff
@@ -28,6 +29,8 @@ module.exports =
 
   activate: ->
     # for debug
+    GitHubApi ?= require 'github'
+    PackageManager ?= require './package-manager'
```
Comment (jerone) [inline lib/sync-settings.coffee]: Can you move these two lines above the debug message, which belongs to the commands below.

Comment (olmokramer) [inline lib/sync-settings.coffee]: Does the comment make sense anyway? It's not like adding the commands is for debugging purposes only right?

Comment (jerone) [inline lib/sync-settings.coffee]: True, if you want you can remove the comment.

Comment (olmokramer) [inline lib/sync-settings.coffee]: Ok, removed it.

**Comment by jerone** [comment]
:+1:

**Pull request merged**